src/middlewared/middlewared/plugins/smb.py

```diff
@@ -902,6 +946,37 @@ async def close_share(self, share_name):
     async def clean(self, data, schema_name, verrors, id=None):
         data['name'] = await self.name_exists(data, schema_name, verrors, id)
 
+
+    @private
+    async def validate_aux_params(self, data, schema_name, verrors):
+        """
+        libsmbconf expects to be provided with key-value pairs.
+        Perform basic normalization and validation of user-provided data.
+        """
+        parsed_auxsmbconf = ""
+        for entry in data['auxsmbconf'].splitlines():
+            if entry == "":
+                # silently remove empty lines
+                continue
+            if entry.startswith('#'):
+                verrors.add(
+                    f'{schema_name}.auxsmbconf',
+                    f'Comments are not permitted in SMB auxiliary parameters: {entry}'
+                )
+                continue
+            kv = entry.split('=', 1)
+            if len(kv) != 2:
+                verrors.add(
+                    f'{schema_name}.auxsmbconf',
+                    f'Auxiliary parameters must be in the format of "key = value": {entry}'
+                )
+                continue
+
+            parsed_auxsmbconf += f'\n{entry}' if len(parsed_auxsmbconf) != 0 else entry
```

Comment:
Isnt this piece of code pretty much the same as the one in validate_smb ?

Perhaps we should "deduplicate" it?